pkg/eks/eks.go

```diff
@@ -228,18 +230,110 @@ func (c *ClusterProvider) WaitForControlPlane(id *api.ClusterMeta, clientSet *ku
+}
+
+func (t *updateClusterConfigTask) Describe() string { return t.info }
+func (t *updateClusterConfigTask) Do(errs chan error) error {
+	err := t.call(t.spec)
+	close(errs)
+	return err
+}
+
+func (c *ClusterProvider) GetClusterConfigForLogging(cl *api.ClusterMeta) (sets.String, sets.String, error) {
+	enabled := sets.NewString()
+	disabled := sets.NewString()
+
+	cluster, err := c.DescribeControlPlane(cl)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, i := range cluster.Logging.ClusterLogging {
+		for _, t := range i.Types {
+			if t == nil {
+				return nil, nil, fmt.Errorf("unexpeted response from EKS API - nil string")
+			}
+			if api.IsEnabled(i.Enabled) {
+				enabled.Insert(*t)
+			}
+			if api.IsDisabled(i.Enabled) {
+				disabled.Insert(*t)
+			}
+		}
+	}
+	return enabled, disabled, nil
+}
+
+// UpdateClusterConfigForLogging calls UpdateClusterConfig to enable logging
+func (c *ClusterProvider) UpdateClusterConfigForLogging(cfg *api.ClusterConfig) error {
+	enabled := sets.NewString(cfg.EnableLogging...)
+	disabled := sets.NewString()
+	for _, facility := range api.SupportedLoggingFacilities() {
+		if !enabled.Has(facility) {
+			disabled.Insert((facility))
+		}
+	}
+
+	logger.Info("updating cluster configuration for logging (enabled: %s & disabled: %s)",
+		strings.Join(enabled.List(), ", "),
+		strings.Join(disabled.List(), ", "),
+	)
+
+	input := &awseks.UpdateClusterConfigInput{
+		Name: &cfg.Metadata.Name,
+		Logging: &awseks.Logging{
+			ClusterLogging: []*awseks.LogSetup{
+				{
+					Enabled: api.Enabled(),
+					Types:   aws.StringSlice(enabled.List()),
+				},
+				{
+					Enabled: api.Disabled(),
+					Types:   aws.StringSlice(disabled.List()),
+				},
+			},
+		},
+	}
+
+	output, err := c.Provider.EKS().UpdateClusterConfig(input)
+	if err != nil {
+		return err
+	}
+	if err := c.waitForUpdateToSucceed(cfg.Metadata.Name, output.Update); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateClusterConfigTasks returns all tasks for updating cluster configuration
+func (c *ClusterProvider) UpdateClusterConfigTasks(cfg *api.ClusterConfig) *manager.TaskTree {
+	tasks := &manager.TaskTree{Parallel: false}
+
+	tasks.Append(&updateClusterConfigTask{
+		info: "update cluster logging configurtaion",
+		spec: cfg,
+		call: c.UpdateClusterConfigForLogging,
+	})
+
+	return tasks
+}
+
 // UpdateClusterVersion calls eks.UpdateClusterVersion and updates to cfg.Metadata.Version,
 // it will return update ID along with an error (if it occurrs)
-func (c *ClusterProvider) UpdateClusterVersion(cfg *api.ClusterConfig) (string, error) {
+func (c *ClusterProvider) UpdateClusterVersion(cfg *api.ClusterConfig) (*awseks.Update, error) {
 	input := &awseks.UpdateClusterVersionInput{
 		Name:    &cfg.Metadata.Name,
 		Version: &cfg.Metadata.Version,
 	}
 	output, err := c.Provider.EKS().UpdateClusterVersion(input)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return *output.Update.Id, nil
+	return output.Update, nil
```

Comment:
It's better to keep our own API's free of third party libraries for testability and maintainability. Can we return the id instead? AFAICT that's the only thing we need